src/components/learner-credit-management/search/CatalogSearch.jsx

```diff
@@ -1,19 +1,17 @@
 import React from 'react';
-import { useParams } from 'react-router-dom';
 import algoliasearch from 'algoliasearch/lite';
 import { Configure, InstantSearch } from 'react-instantsearch-dom';
+import PropTypes from 'prop-types';
 
 import { FormattedMessage } from '@edx/frontend-platform/i18n';
 import { SearchHeader } from '@edx/frontend-enterprise-catalog-search';
 
 import { configuration } from '../../../config';
 import CatalogSearchResults from './CatalogSearchResults';
 
-const CatalogSearch = () => {
-  const { budgetId } = useParams();
+const CatalogSearch = ({ catalogUuid }) => {
```

Comment:
[question] Is it expected that mousing out of the window and back into this component would trigger another more queries to Algolia? As seen below with the "Network" tab open, it looks like there are Algolia `queries` API calls being made when the user's cursor leaves the window and then again when subsequently re-entering. It's especially noticeable when the "Network" tab is throttled to "Fast 3G" or "Slow 3G", as the skeleton loading states become visible.


https://github.com/openedx/frontend-app-admin-portal/assets/2828721/93cc7f45-ed1b-4b31-8c81-c069cffbd4dd

https://github.com/openedx/frontend-app-admin-portal/assets/2828721/d96bc7d5-d890-4112-b9d7-638a0ddc69ea

Reply:
I think this is happening because the page is re-rendering every time the user navigates to the sidebar and back. It looks like this is a larger issue with the app because I'm seeing other pages re-render as well.

Reply:
https://github.com/openedx/frontend-app-admin-portal/assets/71999631/99e92ab5-f005-44f0-9a40-b0245160f900